### PR TITLE
feat: add setup-extension action

### PR DESCRIPTION
**Pull request by pweyck**
feat: add phpstan action

needs https://github.com/shopware/setup-shopware/pull/4

### CODE REVIEW

**Review by mkraeml — APPROVED** [review]
Like we said in the Daily, it would be cool if we could overwrite the `minimum-stability` of shopware, so we could install dev packages using `setup-extension`.
Otherwise everthing looks good.

File: phpstan/action.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    default: ${{ github.workspace }}/var/phpstan
+
+runs:
+  using: "composite"
+  steps:
+    - name: Create phpstan.neon
+      working-directory: custom/plugins/${{ inputs.extensionName }}
+      shell: bash
+      run: |
+        printf "parameters:\n  tmpDir: ${{ inputs.cacheDir }}\nincludes:\n  - phpstan.neon.dist" > phpstan.neon
```
Comment (philipreinken) [inline phpstan/action.yml:23]: Aren't we ignoring any future additions/changes in `phpstan.neon` this way?. Maybe configuring the cache can be achieved using `sed` or something, just replacing the `tmpDir`.

**Review by philipreinken — APPROVED** [review]

File: phpstan/action.yml (hunk 1 above)
Comment (pweyck) [reply phpstan/action.yml:23]: If there's no phpstan.neon, it should work. I'll check our plugins, if all of them use phpstan.neon.dist, it should work. We could also add check that the file does not exist yet

Comment (philipreinken) [reply phpstan/action.yml:23]: > We could also add check that the file does not exist yet

Sounds good 👍 As long as there's no proper way to decode/encode `neon` using a standalone binary or standard CLI tooling, it's also the only option I suppose.

File: setup-extension/action.yml [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+runs:
+  using: "composite"
+  steps:
+    - uses: shopware/github-actions/shopware-version@main
+      with:
+        fallback: ${{ inputs.shopwareVersionFallback }}
+      id: version
+      if: ${{ inputs.shopwareVersion == '.auto' }}
+    - name: Setup Shopware
+      uses: shopware/setup-shopware@fix/hashFiles
```
Comment (pweyck) [inline setup-extension/action.yml:76]: merge first

File: setup-extension/action.yml [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        composer config minimum-stability dev
+
+        AUTOLOAD_DEV="$(jq '."autoload-dev"."psr-4" | to_entries | map({"key": .key, "value": "custom/plugins/${{ inputs.extensionName }}/\(.value)"}) | from_entries' < custom/plugins/${{ inputs.extensionName }}/composer.json)"
+        jq -s '.[0] * {"repositories": .[1]} * {"autoload-dev": {"psr-4": .[2]}}' composer.json <(echo '${{ inputs.extraRepositories }}') <(echo "$AUTOLOAD_DEV") > composer.json.new
+        mv composer.json.new composer.json
+
+    - name: Add extra repositories
+      if: ${{ inputs.extraRepositories }}
+      shell: bash
+      run: |
```
Comment (shyim) [inline setup-extension/action.yml:104]: seems useless this step 🤔

Comment (pweyck) [reply setup-extension/action.yml:104]: currently, we need this for rufus

**Review by mkraeml — APPROVED** [review]

**Pull request merged**